web/source/utils/version.ts

```diff
@@ -0,0 +1,77 @@
+namespace com.keyman.utils {
+  // Dotted-decimal version
+  export class Version {
+    public static readonly NO_DEFAULT_KEYCAPS = new Version("12.0");
+
+    private readonly components: number[]
+
+    constructor(text: String) {
```

Comment:
You could have an alternate which is passed an array of integers to avoid having to parse it, e.g. `Version([12,0])` when passed as a constant from code.